src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js

```diff
@@ -38,9 +35,13 @@ export const EnrollmentAddEventPage = () => {
         dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
     }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
 
-    const handleSave = useCallback(() => {
-        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
-    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+    const handleSave = useCallback(
+        (data, uid) => {
+            uid && dispatch(updateEnrollmentEventsWithoutId(uid, data.events[0]));
```

Comment:
Can we annotate the `uid` instead of doing `uid && ...`?
```suggestion
        (data, uid: string) => {
           dispatch(updateEnrollmentEventsWithoutId(uid, data.events[0]));
```